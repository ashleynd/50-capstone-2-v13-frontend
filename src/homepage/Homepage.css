.Homepage {
  /* put around the vertical center of page */
  height: calc(100vh - 150px);
  display: flex;
  align-items: center;
}

.listofquotes {
  margin: 100px;
  margin-top: -90px;
}

.lead {
  margin-top: 30px;
  font-size: x-large;
}

.headline {
  font-size: 37px;
}