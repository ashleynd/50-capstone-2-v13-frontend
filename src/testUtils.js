import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedAQuote = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedAQuote }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
