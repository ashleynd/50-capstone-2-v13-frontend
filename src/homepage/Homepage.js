import React, { useContext } from "react";
// import { Link } from "react-router-dom";
import "./Homepage.css";
import UserContext from "../auth/UserContext";
// import RandomQuote from "../RandomQuote/RandomQuote";
import MainQuoteList from "../mainquotelist/MainQuoteList";

/** Homepage of site.
 *
 * Shows welcome message or login/register buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

function Homepage() {
  const { currentUser } = useContext(UserContext);
  console.debug("Homepage", "currentUser=", currentUser);

  return (
      <div className="App">
        <div className="container text-center">
          <h1 className="headline">Quotable</h1>
          {/* <p className="lead">Browse quotes from popular authors.</p> */}
          <p className="lead">Join or login to view and like more famous quotes, and post your own quotes.</p>
          {currentUser
              ? <h2>
                Welcome Back, {currentUser.firstName || currentUser.username}!
                {/* <RandomQuote /> */}
              </h2>
              : (
                <p>
                    {/* <Link className="btn btn-primary font-weight-bold mr-3"
                          to="/login">
                      Log in
                    </Link>
                    <Link className="btn btn-primary font-weight-bold"
                          to="/signup">
                      Sign up
                    </Link> */}
                  </p>
              )}
              <div className="listofquotes" style={{ display: "flex", flexWrap: "wrap" }}>
              <MainQuoteList />
              </div>
        </div>
      </div>
  );
}

export default Homepage;
