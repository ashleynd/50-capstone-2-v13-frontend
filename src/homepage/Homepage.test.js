import React from "react";
import { render } from "@testing-library/react";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
      <UserProvider>
        <Homepage />
      </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

