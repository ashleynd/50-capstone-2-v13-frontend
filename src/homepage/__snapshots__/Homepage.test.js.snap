// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="container text-center"
    >
      <h1
        class="headline"
      >
        Quotable
      </h1>
      <p
        class="lead"
      >
        Join or login to view and like more famous quotes, and post your own quotes.
      </p>
      <h2>
        Welcome Back, testuser!
      </h2>
      <div
        class="listofquotes"
        style="display: flex; flex-wrap: wrap;"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="container text-center"
    >
      <h1
        class="headline"
      >
        Quotable
      </h1>
      <p
        class="lead"
      >
        Join or login to view and like more famous quotes, and post your own quotes.
      </p>
      <p />
      <div
        class="listofquotes"
        style="display: flex; flex-wrap: wrap;"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
